### PR TITLE
fix: Bugfixes for detected  labels API

### PR DESCRIPTION
**Pull request by shantanualsi**
**What this PR does / why we need it**:
Fixes bug in the detected labels API. 

- Query does not work as matches are not recognized
- Blank value increases the cardinality by 1
- Dedupe does not work properly
- panics if ingester labels are nil

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -1403,7 +1403,7 @@ func (i *Ingester) GetDetectedLabels(ctx context.Context, req *logproto.Detected
 	}
 	result := make(map[string]*logproto.UniqueLabelValues)
 	for label, values := range labelMap {
-		uniqueValues := make([]string, len(values))
```
Comment (shantanualsi) [inline pkg/ingester/ingester.go:1406]: use of make introduced blank values in the slice

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -1389,7 +1389,7 @@ func (i *Ingester) GetDetectedLabels(ctx context.Context, req *logproto.Detected
 	}
 	var matchers []*labels.Matcher
 	if req.Query != "" {
-		matchers, err := syntax.ParseMatchers(req.Query, true)
```
Comment (shantanualsi) [inline pkg/ingester/ingester.go:1392]: := again after declaring zero value above 🧠

File: pkg/querier/ingester_querier.go [hunk 3]
```diff
@@ -370,17 +370,24 @@ func (q *IngesterQuerier) DetectedLabel(ctx context.Context, req *logproto.Detec
...
+				combinedValues = thisIngesterValues.Values
+			}
+			labelMap[label] = combinedValues
 		}
 	}
+
+	// Dedupe all ingester values
 	mergedResult := make(map[string]*logproto.UniqueLabelValues)
 	for label, val := range labelMap {
-		uniqueValues := slices.CompactFunc(val, strings.EqualFold)
```
Comment (shantanualsi) [inline pkg/querier/ingester_querier.go:383]: Compact only works for dedupe if values are sorted

**Review by cyriltovena — APPROVED** [review]
LGTM

Many bug fixes, few tests proving it's fixes. Please add more in follow up PR.

**Pull request merged**